launch_ros/launch_ros/parameter_descriptions.py

```diff
@@ -154,3 +163,96 @@ def evaluate(self, context: LaunchContext) -> Tuple[Text, 'EvaluatedParameterVal
         self.__evaluated_parameter_name = name
         self.__evaluated_parameter_rule = (name, value)
         return (name, value)
+
+
+class ParameterFile:
+    """Describes a ROS parameter file."""
+
+    def __init__(
+        self,
+        param_file: Union[FilePath, SomeSubstitutionsType],
+        *,
+        allow_substs: bool = False
+    ) -> None:
+        """
+        Construct a parameter file description.
+
+        :param param_file: Path to a parameter file.
+        :param allow_subst: Allow substitutions in the parameter file.
+        """
+        ensure_argument_type(
+            param_file,
+            SomeSubstitutionsType_types_tuple + (os.PathLike, bytes),
+            'param_file',
+            'ParameterFile()'
+        )
+        ensure_argument_type(
+            allow_substs,
+            bool,
+            'allow_subst',
+            'ParameterFile()'
+        )
+        self.__param_file: Union[List[Substitution], FilePath] = param_file
+        if isinstance(param_file, SomeSubstitutionsType_types_tuple):
+            self.__param_file = normalize_to_list_of_substitutions(param_file)
+        self.__allow_substs = allow_substs
+        self.__evaluated_param_file: Optional[Path] = None
+        self.__created_tmp_file = False
+
+    @property
+    def param_file(self) -> Union[FilePath, List[Substitution]]:
+        """Getter for parameter file."""
+        if self.__evaluated_param_file is not None:
+            return self.__evaluated_param_file
+        return self.__param_file
+
+    @property
+    def allow_substs(self) -> bool:
+        """Getter for allow substitutions argument."""
+        return self.__allow_substs
+
+    def __str__(self) -> Text:
+        return (
+            'launch_ros.description.ParameterFile'
+            f'(param_file={self.param_file}, allow_substs={self.allow_substs})'
+        )
+
+    def evaluate(self, context: LaunchContext) -> Path:
+        """Evaluate and return a parameter file path."""
+        if self.__evaluated_param_file is not None:
+            return self.__evaluated_param_file
+
+        param_file = self.__param_file
+        if isinstance(param_file, list):
+            # list of substitutions
+            param_file = perform_substitutions(context, self.__param_file)
+        param_file_path: Path = Path(param_file)
+        if self.__allow_substs:
+            with ExitStack() as stack:
+                f = stack.enter_context(open(param_file_path, 'r'))
+                h = stack.enter_context(
+                    NamedTemporaryFile(mode='w', prefix='launch_params_', delete=False))
+                read = f.read()
+                parsed = perform_substitutions(context, parse_substitution(read))
```

Comment:
@ivanpauno nit^N: collapse the last two lines into one.

Reply:
https://github.com/ros2/launch_ros/pull/168/commits/98011d9ed7915824ae415e56444dcc29d22aea53